tests/test_image_upload.py

```diff
@@ -150,3 +151,152 @@ def test_upload_image_invalid_location_header(self):
 
         with self.assertRaisesRegexp(ImageUploadException, 'S3 Invalid location header'):
             s3.upload(img_data, 'image4.png')
+
+    @responses.activate
+    def test_returns_object_url_when_aws_url_with_201_status_code(self):
+        mock_request(status=201, fixture='s3_success_response.xml',
+                     include_location=False)
+
+        signed_request = get_signed_request()
+        img_data = get_fixture('image1.png', as_binary=True)
+
+        s3 = S3ImageUploader(signed_request)
+        result = s3.upload(img_data, 'image1.png', 'png')
+
+        self.assertEqual('http://pydocx.s3.amazonaws.com/uploads/pydocx/image1.png', result)
+
+class GCSImageUploaderTestCase(TestCase):
+    @responses.activate
+    def test_content_type_not_set_when_gcs_url(self):
+        filename = 'image1.png'
+        image_data = get_fixture(filename, as_binary=True)
+        xml_body = get_fixture('gcs_success_response.xml', as_binary=True)
+
+        signed_request = get_signed_request()
+        signed_request['url'] = 'https://pydocx.storage.googleapis.com'
+
+        signed_mock = get_signed_request()
+        signed_mock.pop('url')
+        files = {"file": (filename, image_data)}
+        responses.add(responses.POST,
+                      signed_request['url'],
+                      status=201,
+                      match=[multipart_matcher(files, data=signed_mock)],
+                      content_type='multipart/form-data',
+                      body=xml_body)
```

Comment:
NON-BLOCKING: Would it be relatively straightforward/make sense to add a `gcs_mock_request` utility?

Reply:
Yes. I will toil in the python pit once more.

Reply:
576b1216d397b0c03427000b8f03b7e841afc011